planning/behavior_velocity_planner/autoware_behavior_velocity_roundabout_module/src/scene_roundabout_prepare_data.cpp

```diff
@@ -0,0 +1,385 @@
+    }
+  }
+
+  // final objective lanelets
+  lanelet::ConstLanelets conflicting_ex_ego_yield_lanelets;
+  lanelet::ConstLanelets conflicting_ex_ego_lanelets;
+  // conflicting lanes is necessary to get stopline for stuck vehicle
+  for (auto && conflicting_lanelet : conflicting_lanelets) {
+    if (!lanelet::utils::contains(ego_lanelets, conflicting_lanelet))
+      conflicting_ex_ego_lanelets.push_back(conflicting_lanelet);
+  }
+
+  // exclude yield lanelets and ego lanelets from conflicting_lanelet
+  for (const auto & conflicting_lanelet : conflicting_lanelets) {
+    if (
+      lanelet::utils::contains(yield_lanelets, conflicting_lanelet) ||
+      lanelet::utils::contains(ego_lanelets, conflicting_lanelet)) {
+      continue;
+    }
+    conflicting_ex_ego_yield_lanelets.push_back(conflicting_lanelet);
+  }
+
+  // get possible lanelet path that reaches conflicting_lane longer than given length
+  lanelet::ConstLanelets detection_and_preceding_lanelets;
+  {
+    std::set<lanelet::Id> detection_ids;
+    for (const auto & ll : conflicting_ex_ego_yield_lanelets) {
+      // Preceding lanes does not include detection_lane so add them at the end
+      const auto & inserted = detection_ids.insert(ll.id());
+      if (inserted.second) detection_and_preceding_lanelets.push_back(ll);
+      // get preceding lanelets without ego_lanelets
+      // to prevent the detection area from including the ego lanes and its' preceding lanes.
+      const auto lanelet_sequences = lanelet::utils::query::getPrecedingLaneletSequences(
+        routing_graph_ptr, ll, detection_area_length, ego_lanelets);
+      for (const auto & ls : lanelet_sequences) {
+        for (const auto & l : ls) {
+          const auto & inner_inserted = detection_ids.insert(l.id());
+          for (const auto & roundabout : roundabout_regulatory_elements) {
+            if (
+              inner_inserted.second && roundabout->isRoundaboutLanelet(l.id()) &&
+              !lanelet::utils::contains(associative_ids_, l.id()))
+              detection_and_preceding_lanelets.push_back(l);
+          }
+        }
+      }
+    }
+  }
+
+  auto [attention_lanelets, original_attention_lanelet_sequences] =
+    util::mergeLaneletsByTopologicalSort(
+      detection_and_preceding_lanelets, conflicting_ex_ego_yield_lanelets, routing_graph_ptr);
+
+  RoundaboutLanelets result;
+  result.attention_ = std::move(attention_lanelets);
+  result.attention_non_preceding_ = std::move(conflicting_ex_ego_yield_lanelets);
+  result.conflicting_ = std::move(conflicting_ex_ego_lanelets);
+  result.adjacent_ = planning_utils::getConstLaneletsFromIds(lanelet_map_ptr, associative_ids_);
+
+  // NOTE: to properly update(), each element in conflicting_/conflicting_area_,
+  // attention_non_preceding_/attention_non_preceding_area_ need to be matched
+  result.attention_area_ = util::getPolygon3dFromLanelets(result.attention_);
+  result.attention_non_preceding_area_ =
```

Comment:
Although this function is complicated, why not we just create attention_area from the lanelets saved in rountdabout regulatory element ? then this function will be in a few lines.

The thing is, currently `attention_area_length` parameter does not do anything at all